src/deploy/functions/cache/applyHash.ts

```diff
@@ -0,0 +1,56 @@
+import { Backend, Endpoint, allEndpoints } from "../backend";
+import * as args from "../args";
+import {
+  getEndpointHash,
+  getEnvironmentVariablesHash,
+  getSecretsHash,
+  getSourceHash,
+} from "./hash";
+
+/**
+ *
+ * Updates all the CodeBase {@link Backend}, applying a hash to each of their {@link Endpoint}.
+ */
+export async function applyBackendHashToBackends(
+  wantBackends: Record<string, Backend>,
+  context: args.Context
+): Promise<void> {
+  // envHash
+  const envHash = getEnvHash(wantBackends);
```

Comment:
I think this logic is wrong. Each codebase has its own set of environment variables, optionally specified by `.env` in its source directory.

Reply:
Thanks for calling that out. Its safer to repeat the calls for those instances. Updated!